docs/architecture/deployment-execution/import.md

```diff
@@ -0,0 +1,184 @@
+newer method of importing resources into a stack that also generates program
+code for imported resources. `pulumi import` accepts a list of *import specs*,
+where each spec comprises at minimum a [type token](urns), name, and
+[ID](resource-ids), but may also specify a parent URN, provider reference, and
+package version. Unlike the `import` resource option, `pulumi import` does not
+insist that the desired state of the resource in the Pulumi program matches the
+actual state of the resource as returned by the provider, since it is capable of
+generating code to match the actual state. Given a resource `R` of type `T` with
+import ID `X` and an (initiall empty) set of input properties `Iₚ`, the engine
+performs the following sequence of operations:
+
+1. Fetch the current inputs `Iₐ` and state `Sₐ` for the resource of type `T`
+   with ID `X` from its provider by calling the provider's
+   [](pulumirpc.ResourceProvider.Read) method. If the provider does not return a
+   value for `Iₐ`, the provider does not support importing resources and the
+   import fails.
+2. Fetch the schema for resources of type `T` from the provider. If the provider
+   is not schematized or if `T` has no schema, the import fails.
+3. Copy the value of each required input property defined in the schema for `T`
+   from `Iₐ` to `Iₚ`.
+4. Validate the resource's inputs and apply any programmatic defaults by passing
+   `Iₚ` and `Iₐ` to the provider's [](pulumirpc.ResourceProvider.Check) method.
+   Let `Iₖ` be the checked inputs; these inputs form the resource's desired
+   state.
+5. Check for differences between `Iₖ` and `Sₐ` by calling the provider's
+   [](pulumirpc.ResourceProvider.Diff) method. If the provider reports any
+   differences, the values of the differing properties are copied from `Sₐ` to
+   `Iₚ`. This is intended to produce the smallest valid set of inputs necessary
+   to avoid diffs. This does not use a fixed-point algorithm because there is no
+   guarantee that the values copied from `Sₐ` are in fact valid (state and
+   inputs with the same property paths may have different types and validation
+   rules) and there is no guarantee that such an algorithm would terminate (
+   bridged Terraform providers, for example, have had bugs that cause persistent
+   diffs, which can only be worked around with `ignoreChanges`).
+
+If all of these steps succeed, the user is left with a definition for `R` in
+their state. The Pulumi CLI then passes the inputs `Iₚ` stored in the state to
+the import code generator. The import code generator converts the values present
+in `Iₚ` into an equivalent [PCL](pcl) representation of `R`'s desired state,
+then passes the PCL to a language-specific code generator to emit a
+representation of `R`'s desired state in the language used by the destination
+stack's Pulumi program. The user can then copy the generated definition into
+their Pulumi program.
+
+```{mermaid}
```

Comment:
If this was ```mermaid``` and not `{mermaid}` GitHub would render it in-line I think. But maybe this is needed for primary renderer.

Reply:
Yes, annoyingly it is needed for Sphinx/Myst to render it correctly.